snapcraft/internal/elf.py

```diff
@@ -82,21 +125,42 @@ def patch(self, *, elf_file: ElfFile) -> None:
         :raises snapcraft.internal.errors.PatcherError:
             raised when the elf_file cannot be patched.
         """
-        # When setting rpath for libraries is implemented, we will do more
-        # here.
-        if not elf_file.is_executable:
-            return
+        if elf_file.is_executable:
+            self._patch_interpreter(elf_file)
+        if elf_file.dependencies:
+            self._patch_rpath(elf_file)
+
+    def _patch_interpreter(self, elf_file: ElfFile) -> None:
+        self._run_patchelf(['--set-interpreter',  self._dynamic_linker],
+                           elf_file.path)
+
+    def _patch_rpath(self, elf_file: ElfFile) -> None:
+        rpath = self._get_rpath(elf_file)
+        print(elf_file.path, 'rpath: ', rpath)
+        self._run_patchelf(['--force-rpath', '--set-rpath', rpath],
```

Comment:
Is it asking for problems to add `--shrink-rpath` here? Note that it would be nice to see a blurb about the options you're using here and why.

Reply:
sure, let's add that and document the params here

Reply:
`--shrink-path` is not something we want, it will discard `$ORIGIN` paths in favour of the standard paths, e.g.;
```
$ ldd /home/ubuntu/git/prime/libexec/git-core/git-imap-send|grep curl
	libcurl.so.4 => /usr/lib/x86_64-linux-gnu/libcurl.so.4 (0x00007f02f7b7f000)
```

versus not using `--shrink-path`:
```
$ ldd /home/ubuntu/git/prime/libexec/git-core/git-imap-send|grep curl
	libcurl.so.4 => /home/ubuntu/git/prime/libexec/git-core/../../usr/lib/x86_64-linux-gnu/libcurl.so.4 (0x00007f4dab7bb000)
```